app/src/main/java/com/projectit3048c/MainViewModel.kt

```diff
@@ -16,8 +16,8 @@ class MainViewModel(var foodService : IFoodService =  FoodService()) : ViewModel
     fun fetchFoods() {
         viewModelScope.launch {
            var innerFoods = foodService.fetchFoods()
-            if (innerFoods != null) {
-                foods.postValue(innerFoods!!)
+            innerFoods?.let { nonNullFoods ->
+                foods.postValue(nonNullFoods)
```

Comment:
Updated non-nullable assertion to avoid using !!, referenced from https://kotlinlang.org/docs/java-to-kotlin-nullability-guide.html#default-values-instead-of-null